### PR TITLE
2023 Plan Pricing Grid: Only display tax exclusion copy for EU countries

### PR DESCRIPTION
**Pull request by jeyip**
Related to https://github.com/Automattic/martech/issues/1580#issuecomment-1549926816

## Proposed Changes

* Only display text about tax exclusion to users in EU countries
* Because all translations texts have been translated, we remove any conditions checking for whether or not translations have been localized

### Without tax exclusion text
Screenshot 2023-05-22 at 10 43 36 AM

### With tax exclusion text
Screenshot 2023-05-22 at 10 46 53 AM

## Testing Instructions

1. Check out this branch and `yarn start`
2. Proxy as a user from an EU country
    * p1684772351255899/1684771443.298789-slack-CV2TX2PAN 
    * p1684772456190969/1684771443.298789-slack-CV2TX2PAN
3. Create a new user through a locale specific url https://wordpress.com/start/user/{country_slug}. Ensure that the country slug matches the EU country being proxied
    * Ex. https://wordpress.com/start/user/fr for France
    * Ex. https://wordpress.com/start/user/de for Germany
4. Navigate to the 2023 plans pricing grid
5. Verify that the tax exclusion text is shown in the billing timeframe
6. Repeat steps 2-4 for a non-EU country. Verify that the tax exclusion text _is not_ show in the billing timeframe

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plan-features-2023-grid/components/billing-timeframe.tsx [hunk 1]
```diff
@@ -82,22 +84,11 @@ function usePerMonthDescription( {
 				? formatCurrency( planPrices.rawPrice, currencyCode, { stripZeros: true } )
 				: null;
 
-		// TODO: Remove check once text is translated
-		const displayNewPriceText =
-			isEnglishLocale ||
-			( i18n.hasTranslation( 'per month, %(rawPrice)s billed annually, Excl. Taxes' ) &&
```
Comment (jeyip) [inline client/my-sites/plan-features-2023-grid/components/billing-timeframe.tsx:88]: **Note:**

We no longer need to check whether or not translations have been localized. They were completed on May 16th https://github.com/Automattic/wp-calypso/pull/76738#issuecomment-1550128637

**Review by ddc22 — APPROVED** [review]
LGTM 🚢

**Comment by southp** [comment]
Hmm, I happened to notice that `is-country-in-eu.ts` has been deleted in https://github.com/Automattic/wp-calypso/commit/f5444c029eb84e013816e050449c24c53252982e by @noahtallen . @noahtallen could you help us understand why it is deleted, and which alternative we should be using instead? I'd be grateful :)

**Comment by noahtallen** [comment]
Hi! The linked commit is actually part of a PR which hasn't been merged yet. I'm not familiar with that file unfortunately. Was it removed somewhere else?

**Comment by jeyip** [comment]
Taking a closer look at this tomorrow morning

**Comment by jeyip** [comment]
> Taking a closer look at this tomorrow morning

Taking a bit longer to get back to this but will return to it soon

**Comment by aneeshd16** [comment]
I left a comment [here](https://github.com/Automattic/martech/issues/1580#issuecomment-1576923397) related to this PR that could possibly make us reconsider the change. 🙂

**Comment by jeyip** [comment]
Closing per our discussion in https://github.com/Automattic/martech/issues/1580#issuecomment-1578189200

**Pull request closed without merging**